syntax = "proto3";

package dir_watcher;

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

service FileWatcherService {
    rpc UpdateFile (FileUpdate) returns (google.protobuf.Empty);
    rpc Search (SearchRequest) returns (SearchMatchesResponse);
}

enum FileUpdateType {
    CREATED = 0;
    UPDATED = 1;
    DELETED = 2;
}

message FileUpdate {
    File file = 1;
    string uuid = 2;
    FileUpdateType update_type = 3;
}

message SearchRequest {
    string uuid = 1;
    string search_term = 2;
}

message SearchMatchesResponse {
    string uuid = 1;
    string search_term = 2;
    repeated File files = 3;
}

message File {
    string file_path = 1;
    string hash = 2;
    google.protobuf.Timestamp update_time = 3;
}